plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/BaseIcebergConnectorTest.java

```diff
@@ -3630,6 +3634,38 @@ public void testEmptyDelete()
         assertUpdate("DROP TABLE " + tableName);
     }
 
+    @Test
+    public void testCtasFromAVersionedTable()
+            throws Exception
+    {
+        String sourceTableName = "test_ctas_versioned_source_" + randomTableSuffix();
+        String sinkTableNameVersion = "test_ctas_versioned_sink_" + randomTableSuffix();
+        String sinkTableNameTimestamp = "test_ctas_versioned_sink_" + randomTableSuffix();
+
+        assertUpdate("CREATE TABLE " + sourceTableName + "(an_integer integer)");
+        Thread.sleep(1);
+        assertUpdate("INSERT INTO " + sourceTableName + " VALUES (1), (2), (3)", 3);
+        long afterInsert123SnapshotId = getLatestSnapshotId(sourceTableName);
+        long afterInsert123EpochMillis = getCommittedAtInEpochMilliSeconds(sourceTableName, afterInsert123SnapshotId);
+        Thread.sleep(1);
+        assertUpdate("INSERT INTO " + sourceTableName + " VALUES (4), (5), (6)", 3);
+        long afterInsert456SnapshotId = getLatestSnapshotId(sourceTableName);
+        Thread.sleep(1);
+        assertUpdate("DELETE FROM " + sourceTableName + " WHERE an_integer < 5", 4);
+
+        // Clone the source table at the state before the deletion occurred to a sink table
+        assertUpdate("CREATE TABLE " + sinkTableNameVersion + " AS SELECT * FROM " + sourceTableName + " FOR VERSION AS OF " + afterInsert456SnapshotId, 6);
+        assertUpdate("CREATE TABLE " + sinkTableNameTimestamp + " AS SELECT * FROM " + sourceTableName + " FOR TIMESTAMP AS OF " + timestampLiteral(afterInsert123EpochMillis, 9), 3);
```

Comment:
You could replace your `timestampLiteral` method with a call to the sql function `from_unixtime` and use `to_unixtime` in the other method

Reply:
`unixtime` is expressed in seconds and not in milliseconds. This would mean that we'd need to have the test sleeping for a second.